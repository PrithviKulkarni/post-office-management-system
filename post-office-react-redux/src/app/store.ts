import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
