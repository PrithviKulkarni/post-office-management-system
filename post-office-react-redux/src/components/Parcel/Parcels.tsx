import React from 'react'
import { getAllParcels } from '../../features/ParcelSlice';
import { Accordion } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { useAppSelector } from '../../app/hooks';

const Parcels = () => {

  const parcels = useAppSelector(getAllParcels)

  return (
    <div className="parcels" data-testid="parcels">
      <div>
      <div className='App'>
  
     

        <Accordion defaultActiveKey='0'>
          
          <h3 className="text-success text-center"> <Link to={`/createParcel`}> Add a parcel </Link></h3>
        
        
          <h4 className="text-white">
            {' '}
            <u>Parcels</u>{' '}
          </h4>
          

          {parcels?.map(parcel => (
            <Accordion.Item eventKey={parcel.id}>
              <Accordion.Header>
            
                <h6>Parcel no: {parcel.parcel_no} </h6>
              </Accordion.Header>

              <Accordion.Body className="bg-info">
              <b> Parcel weight(kg): </b> {parcel.parcel_weight} <br/>
              <b> Status: </b> {parcel.status}<br/>
              <b> Estimated delivery time (in days):</b> {parcel.estimated_delivery_days} <br/>


             <div className='text-dark'>
            <Link to={`/updateParcel/${parcel.id}`}> <b>Update </b> </Link> &nbsp;&nbsp;
            <br/>
            <Link to={`/deleteParcel/${parcel.id}`}><b>Delete</b></Link>
            </div>

              </Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion>
      </div>
    </div>
      
    </div>
  )
}

export default Parcels
