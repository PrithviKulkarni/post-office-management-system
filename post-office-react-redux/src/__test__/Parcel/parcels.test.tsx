import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import Parcels from "../../components/Parcel/Parcels";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);
it("renders list Parcel page correctly", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Parcels />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies the List Parcel render", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Parcels />
      </BrowserRouter>
    </Provider>
  );

  expect(screen.getByTestId("parcels")).toHaveClass("parcels");
});
