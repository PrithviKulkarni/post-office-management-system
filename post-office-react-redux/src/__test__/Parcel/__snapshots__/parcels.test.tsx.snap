// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list Parcel page correctly 1`] = `
<DocumentFragment>
  <div
    class="parcels"
    data-testid="parcels"
  >
    <div>
      <div
        class="App"
      >
        <div
          class="accordion"
        >
          <h3
            class="text-success text-center"
          >
             
            <a
              href="/createParcel"
            >
               Add a parcel 
            </a>
          </h3>
          <h4
            class="text-white"
          >
             
            <u>
              Parcels
            </u>
             
          </h4>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
